pkg/pool-weighted/contracts/LBPool.sol

```diff
@@ -0,0 +1,312 @@
+                                    Hook Functions
+    *******************************************************************************/
+
+    /**
+     * @notice Hook to be executed when pool is registered.
+     * @dev Returns true if registration was successful, and false to revert the registration of the pool.
+     * @param pool Address of the pool
+     * @return success True if the hook allowed the registration, false otherwise
+     */
+    function onRegister(
+        address,
+        address pool,
+        TokenConfig[] memory,
+        LiquidityManagement calldata
+    ) public view override onlyVault returns (bool) {
+        // Since in this case the pool is the hook, we don't need to check anything else.
+        // We *could* check that it's two tokens, but better to let that be caught later, as it will fail with a more
+        // descriptive error.
+        return pool == address(this);
+    }
+
+    // Return HookFlags struct that indicates which hooks this contract supports
+    function getHookFlags() public pure override returns (HookFlags memory hookFlags) {
+        // Check whether swaps are enabled in `onBeforeSwap`.
+        hookFlags.shouldCallBeforeSwap = true;
+        // Ensure the caller is the owner, as only the owner can add liquidity.
+        hookFlags.shouldCallBeforeAddLiquidity = true;
+    }
+
+    /**
+     * @notice Check that the caller who initiated the add liquidity operation is the owner.
+     * @dev We first ensure the caller is the standard router, so that we know we can trust the value it returns
+     * from `getSender`.
+     *
+     * @param router The address (usually a router contract) that initiated the add liquidity operation
+     */
+    function onBeforeAddLiquidity(
+        address router,
+        address,
+        AddLiquidityKind,
+        uint256[] memory,
+        uint256,
+        uint256[] memory,
+        bytes memory
+    ) public view override onlyVault returns (bool) {
+        if (router == _TRUSTED_ROUTER) {
+            return IRouterCommon(router).getSender() == owner();
+        }
+
+        revert RouterNotTrusted();
+    }
+
+    /**
+     * @notice Called before a swap to let the pool block swaps if not enabled.
+     * @return success True if the pool has swaps enabled.
+     */
+    function onBeforeSwap(PoolSwapParams calldata, address) public view override onlyVault returns (bool) {
+        return _getPoolSwapEnabled();
+    }
+
+    /*******************************************************************************
+                                  Internal Functions
+    *******************************************************************************/
+
+    // This is unused in this contract, but must be overridden from WeightedPool for consistency.
+    function _getNormalizedWeight(uint256 tokenIndex) internal view virtual override returns (uint256) {
+        if (tokenIndex < _NUM_TOKENS) {
+            return _getNormalizedWeights()[tokenIndex];
```

Comment:
Is the conciseness of this preferable to only calculating one weight?

Reply:
It's mainly to not have the "1 - weight" in multiple places, which is a relatively weak argument. It's not directly called here (but can be invoked from WeightedPool functions). This favors readability over gas, but I don't feel strongly.

Reply:
Sure. I was trying to gas optimize here but don't have a strong preference either. We can leave it as is